lizrd/train/scheduler.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC
+import math
+
+from torch.optim import Optimizer
+
+
+def get_scheduler(args):
+    if args.scheduler == "constant":
+        return ConstantScheduler(
+            lr_warmup_steps=args.lr_warmup_steps, lr=args.learning_rate
+        )
+    elif args.scheduler == "cosine":
+        return CosineScheduler(
+            lr_warmup_steps=args.lr_warmup_steps,
+            lr=args.learning_rate,
+            final_lr_step=args.final_lr_step,
+            final_lr_fraction=args.final_lr_fraction,
+        )
+    else:
+        raise ValueError(f"Unknown scheduler: {args.scheduler}")
+
+
+class BaseLRScheduler(ABC):
```

Comment:
Can we rename this to AbstractLRScheduler? We use this convention all over the place